packages/fury-adapter-oas3-parser/lib/parser/oas/parsePathItemObject.js

```diff
@@ -150,6 +152,11 @@ function parsePathItemObject(context, member) {
       resource.href = hrefFromParameters(member.key, parameters);
       resource.hrefVariables = hrefVariablesFromParameters(namespace, parameters);
 
+      const hosts = pathItem.get('servers');
+      if (hosts) {
+        resource.push(hosts);
+      }
+
```

Comment:
The host resources should be stored in the "hosts" attribute as per the design in https://github.com/apiaryio/api-elements/commit/2f9cebfc7d3b3a710172aeb291c4533706352377.

You've added a setter for [`hosts` on the Resource element](https://github.com/apiaryio/api-elements.js/blob/10841673ef9cd813ac9c1db7a57c1bc71765f5a4/packages/api-elements/lib/elements/Resource.js#L28) which I think the intention is that it can be used with something along the lines of:

```js
resource.hosts = pathItem.get('servers');
```

Since `get` should return undefined, I'd imagine that the setter for hosts is a no-op so you can omit checks.

Reply:
@kylef 
> You've added a setter for hosts on the Resource element which I think the intention is that it can be used with something along the lines of:

is the `hosts` already added in the server array classes?
https://github.com/apiaryio/api-elements.js/blob/10841673ef9cd813ac9c1db7a57c1bc71765f5a4/packages/fury-adapter-oas3-parser/lib/parser/oas/parseServersArray.js#L23

> Since get should return undefined, I'd imagine that the setter for hosts is a no-op so you can omit checks.

Do you mean that I can remove some of the test?

Reply:
> Do you mean that I can remove some of the test?

I meant the `if (hosts)` check is perhaps not needed (providing setter works as intended with setting undefined`.

Reply:
> isn't the hosts already added in the server array classes?

It is, but I guess it isn't needed, in that case you could set the `hosts` as array element via something like:

```js
const hosts = pathItem.get('servers');
if (hosts) {
  resource.hosts = hosts.content;
}
```

That way the `hosts` class and category element is discarded here.

Reply:
ok so in `parsePathItemObject.js` I set the import of the ServersArray as such:

```
resource.hosts = pathItem.get('servers');
```

If you want me to remove the `hosts` `classes` from `parseServersArray.js` tell me